### ISSUE TITLE
Error on ZoomUs.joinMeeting: "[Error: Error: 4, internalErrorCode=0]"

### ISSUE THREAD

**Issue by tommi97** [mieszko4/react-native-zoom-us#272]
In some Android devices I have the same error when i try to join or start a meeting, with both version 6.15.1 and 6.16.5.
IOS work fine.

When i try to join a meeting I recive following event before the "error 4":
{"event": "meetingClientIncompatible", "status": "MEETING_STATUS_FAILED"}

How can i resolve this issue in Android?

**Comment by poojabisht2202** [comment]
Facing the very same issue from past 1 week

**Comment by poojabisht2202** [comment]
Is it related to sdk keys ?

**Comment by tommi97** [comment]
no, sdk keys are correct.

6.15.1 version is too old but when I try to use 6.16.5 it will crash the entire app when I use joinMeeting/startMeeting function

**Comment by poojabisht2202** [comment]
My app is getting crashed on both versions. Any other update on the issue ?

**Comment by tommi97** [comment]
The problem was that the zoom SDK used form the package is too old, so I decide to use this guide: [https://stefan-majiros.com/blog/integrating-zoom-sdk-into-react-native/](https://stefan-majiros.com/blog/integrating-zoom-sdk-into-react-native/) to implement my custom bridge from React and Zoom SDK to be free to update the zoom SDK.

After following this guide I found some difficulty because the code doesn't work correctly but after some modifications now it works.

When it finally worked, I changed the startMeeting function to accept ZAK authentication instead of JWT because it will be deprecated.

Finally I copied from this package ([react-native-zoom-us](https://github.com/mieszko4/react-native-zoom-us)) some pieces of code to implement some parameters that I used before like: disableShowVideoPreviewWhenJoinMeeting, language and so on...

I hope I was helpful, Good luck! :muscle:
